src/sms-verification/scripts/directives.js

```diff
@@ -0,0 +1,64 @@
+(function (angular) {
+    'use strict';
+
+    var module = angular.module('user_management.sms-verification');
+
+    module.directive('smsVerify', [
+        '$parse',
+        '$routeParams',
+        'smsVerificationFactory',
+        function ($parse, $routeParams, smsVerificationFactory) {
+            return {
+                restrict: 'A',
+                scope: true,
+                templateUrl: 'templates/user_management/sms-verification/verify_form.html',
+                link: function (scope, element, attrs) {
+
+                    // 2-way bind data attribute to scope.
+                    // scope: { data: '=' }
+                    if (angular.isDefined(attrs.data)) {
```

Comment:
It does seem better to have a two-way bound scope using `{ data: '=?' }` because otherwise we are essentially re-implementing built-in functionality of angular for the two-way binding. 

That seems a better solution unless there's a specific reason you don't want to totally isolate the scope inside this directive? Since it uses it's own template, it seems to make sense to isolate the scope that way.